pyiron_base/storage/datacontainer.py

```diff
@@ -837,10 +838,17 @@ def normalize_key(name):
                     (*normalize_key(n), hdf[n] if not self._lazy else HDFStub(hdf, n))
                 )
             for g in hdf.list_groups():
+
+                def to_object(hdf_group):
+                    if isinstance(hdf_group, ProjectHDFio):
+                        return hdf_group.to_object()
+                    else:
+                        return hdf_group
+
                 items.append(
                     (
                         *normalize_key(g),
-                        hdf[g].to_object() if not self._lazy else HDFStub(hdf, g),
+                        to_object(hdf[g]) if not self._lazy else HDFStub(hdf, g),
```

Comment:
I've just noticed it, but this logic also needs to be applied in [HDFStub](https://github.com/pyiron/pyiron_base/blob/9dec6f9744b937548d8846bf4226520bae3b98e4/pyiron_base/storage/hdfstub.py#L101) otherwise lazy loading will not work.

Reply:
Everything else lgtm, though.